### PR TITLE
Add image to doughnuts

### PR DESCRIPTION
**Pull request by irismoini**
- added image_url field to doughnut model and made minor modifications to backend to reflect this change
- added image path field to edit/upload form
- got images to display based on image path provided by user

### CODE REVIEW

**Comment by irismoini** [comment]
Potential Todo: add additional verifications for the image path provided by the user.

If anyone has any suggestions on how to check to see if a url is a valid image path lmk!

File: db/schema.rb [hunk 1]
```diff
@@ -66,7 +67,7 @@
     t.datetime "updated_at", precision: 6, null: false
   end
 
-  add_foreign_key "cart_items", "carts", on_delete: :cascade
```
Comment (chrisyang365) [inline db/schema.rb]: Revert this change. It's necessary to ensure that clearing a `Cart` works properly

**Comment by chrisyang365** [comment]
> Potential Todo: add additional verifications for the image path provided by the user.
> 
> If anyone has any suggestions on how to check to see if a url is a valid image path lmk!

I don't think you necessarily need to verify that the image URL is valid on the backend side. You can just display whatever image that the image URL points to on the frontend like you currently are, but if the image URL isn't valid, the `` attribute should have an `alt` prop that you can insert to specify what should be displayed, in the event that the imageURL specified in the `src` prop is not valid.

**Comment by andrewhlu** [comment]
> > Potential Todo: add additional verifications for the image path provided by the user.
> > If anyone has any suggestions on how to check to see if a url is a valid image path lmk!
> 
> I don't think you necessarily need to verify that the image URL is valid on the backend side. You can just display whatever image that the image URL points to on the frontend like you currently are, but if the image URL isn't valid, the `` attribute should have an `alt` prop that you can insert to specify what should be displayed, in the event that the imageURL specified in the `src` prop is not valid.

I don't think you can either, since there's always the possibility that the content at the URL changes after you submit it with the form at the first time. So any checks should be done at client-side rendering time.

**Comment by aroddick** [comment]
LGTM just make sure you address @chrisyang365 comment
